components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java

```diff
@@ -470,22 +470,48 @@ public Response apisApiIdDocumentsPost(String apiId, DocumentDTO body, String co
         try {
             APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
             Documentation documentation = DocumentationMappingUtil.fromDTOtoDocumentation(body);
+            String documentName = body.getName();
             String tenantDomain = RestApiUtil.getLoggedInUserTenantDomain();
 
+            if (body.getType() == DocumentDTO.TypeEnum.OTHER && StringUtils.isBlank(body.getOtherTypeName())) {
+                //check otherTypeName for not null if doc type is OTHER
+                throw RestApiUtil.buildBadRequestException("otherTypeName cannot be empty if type is OTHER.");
+            }
+
+            String sourceUrl = body.getSourceUrl();
+            if (body.getSourceType() == DocumentDTO.SourceTypeEnum.URL &&
+                    (StringUtils.isBlank(sourceUrl) || !RestApiUtil.isURL(sourceUrl))) {
+                throw RestApiUtil.buildBadRequestException("Invalid document sourceUrl Format");
+            }
+
```

Comment:
Fixed in #1581 (https://github.com/wso2/carbon-apimgt/commit/93750d080d7a4e1c2aa7e7fcf195e110234d3b9b)